dbt/redshift/redshift_etl.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import absolute_import
+import logging
+import os
+import psycopg2
+import subprocess
+from dateutil import rrule
+
+from parsely_raw_data import redshift as parsely_redshift
+from parsely_raw_data import utils as parsely_utils
+from dbt.redshift.settings.default import *
+from dbt.redshift.settings.merge_settings_yaml import migrate_settings
+
+SETTINGS_ARG_MAPPING = {
+    'table_name': PARSELY_RAW_DATA_TABLE,
+    'host': REDSHIFT_HOST,
+    'user': REDSHIFT_USER,
+    'password': REDSHIFT_PASSWORD,
+    'database': REDSHIFT_DATABASE,
+    'port': REDSHIFT_PORT,
+    'keep_extra_data': ETL_KEEP_RAW_DATA,
+    'network': S3_NETWORK_NAME,
+    'access_key_id': S3_AWS_ACCESS_KEY_ID,
+    'secret_access_key': S3_AWS_SECRET_ACCESS_KEY,
+    'start_date': ETL_START_DATE,
+    'end_date': ETL_END_DATE,
+    'dbt_profiles_dir': DBT_PROFILE_LOCATION,
+    'dbt_target': DBT_PROFILE_TARGET_NAME
+}
+
+
+def get_settings_arg_mapping_value(field_name, arg_value):
+    return arg_value or SETTINGS_ARG_MAPPING[field_name]
+
+
+def migrate_from_s3_by_day(network=S3_NETWORK_NAME,
+                table_name=PARSELY_RAW_DATA_TABLE,
+                host=REDSHIFT_HOST,
+                user=REDSHIFT_USER,
+                password=REDSHIFT_PASSWORD,
+                database=REDSHIFT_DATABASE,
+                port=REDSHIFT_PORT,
+                access_key_id=S3_AWS_ACCESS_KEY_ID,
+                secret_access_key=S3_AWS_SECRET_ACCESS_KEY,
+                start_date=ETL_START_DATE,
+                end_date=ETL_END_DATE,
+                dbt_profiles_dir=DBT_PROFILE_LOCATION,
+                dbt_target=DBT_PROFILE_TARGET_NAME):
+
+    start_date = parsely_utils.parse_datetime_arg(start_date).date()
+    end_date = parsely_utils.parse_datetime_arg(end_date).date()
+
+    # This runs the copy_from_s3 for every day between the start_date and end_date
+    for d in rrule.rrule(rrule.DAILY, interval=1, dtstart=start_date, until=end_date):
+        prefix = 'events/'+ d.strftime('%Y/%m/%d')
+        parsely_redshift.copy_from_s3(network=network,
+                                s3_prefix=prefix,
+                                table_name=table_name,
+                                host=host,
+                                user=user,
+                                password=password,
+                                database=database,
+                                port=port,
+                                access_key_id=access_key_id,
+                                secret_access_key=secret_access_key)
+
+    # This runs dbt once all of the new data has been copied into the raw data table
+    dpl_wd = os.path.join(os.getcwd(), 'dbt/redshift/')
+    subprocess.call(dpl_wd + "run_parsely_dpl.sh " + dbt_profiles_dir + ' ' + dbt_target, shell=True, cwd=dpl_wd)
```

Comment:
Usability nitpick: forming this command and printing it to the console (even if only under certain circumstances) might help with debugging. It puts more info in front of the user, and makes your life easier if something breaks down the road.

Reply:
will do - i'll add logging.info()